x/mongo/driver/topology/topology.go

```diff
@@ -312,6 +313,11 @@ func (t *Topology) SelectServer(ctx context.Context, ss description.ServerSelect
 		var suitable []description.Server
 		var selectErr error
 
+		desc := t.Description()
```

Comment:
Sorry if my comment wasn't clear. I meant that we should remove this check from here and only have it in `selectServerFromDescription` because that function will always be called.

Reply:
I meant to remove this, looks like I forgot. updated now